cmd/preupgradechecks/preupgradechecks.go

```diff
@@ -69,14 +70,30 @@ func (client *PreUpgradeTaskClient) IsFissionReInstall() bool {
 	for i := 0; i < maxRetries; i++ {
 		_, err := client.apiExtClient.ApiextensionsV1beta1().CustomResourceDefinitions().Get(FunctionCRD, metav1.GetOptions{})
 		if err != nil && k8serrors.IsNotFound(err) {
-			return false
-
+			continue
 		}
+		return true
 	}
-
 	return false
 }
 
+// LatestSchemaApplied ensures that the end user has applied the latest CRDs generated to the cluster.
+// For future reference: whenever a new field is added, we need to check for that field's existence in this function
+func (client *PreUpgradeTaskClient) LatestSchemaApplied() error {
+	client.logger.Info("Checking if user has applied the latest CRDs")
+	crd, err := client.apiExtClient.ApiextensionsV1().CustomResourceDefinitions().Get("functions.fission.io", metav1.GetOptions{})
```

Comment:
This line is already getting called. Can we just rename `IsFissionReInstall` to `IsFisisionInstalled` and reuse function here ?
Also `functions.fission.io` is already defined with variable `FunctionCRD`